examples/kg-chat/csv-chat.py

```diff
@@ -0,0 +1,160 @@
+    tools: bool = typer.Option(
+        False, "--tools", "-t", help="use langroid tools instead of function-calling"
+    ),
+    nocache: bool = typer.Option(False, "--nocache", "-nc", help="don't use cache"),
+) -> None:
+    set_global(
+        Settings(
+            debug=debug,
+            cache=nocache,
+            stream=not no_stream,
+        )
+    )
+    print(
+        """
+        [blue]Welcome to CSV Knowledge Graph RAG chatbot!
+        Enter x or q to quit at any point.
+        """
+    )
+
+    load_dotenv()
+
+    neo4j_settings = Neo4jSettings()
+
+    csv_kg_chat_agent = CSVChatGraphAgent(
+        config=Neo4jChatAgentConfig(
+            neo4j_settings=neo4j_settings,
+            use_tools=tools,
+            use_functions_api=not tools,
+            llm=OpenAIGPTConfig(
+                chat_model=model or OpenAIChatModel.GPT4_TURBO,
+                chat_context_length=16_000,  # adjust based on model
+                temperature=0.2,
+                timeout=45,
+            ),
+        ),
+    )
+
+    build_kg_instructions = ""
+
+    buid_kg = Prompt.ask(
+        "Do you want to build the graph database from a CSV file? (y/n)",
+        default="y",
+    )
+    if buid_kg == "y":
+        csv_location = Prompt.ask(
+            "Please provide the path/URL to the CSV",
+            default="examples/docqa/data/imdb-drama.csv",
+        )
+
+        csv_dataframe = _load_csv_dataset(csv_location)
+        headers = csv_dataframe.columns.tolist()
+
+        # clean the CSV file before loading it into Neo4j
+        csv_dataframe = _preprocess_dataframe_for_neo4j(csv_dataframe)
+
+        num_rows = len(csv_dataframe)
+
+        if num_rows > 1000:
+            print(
+                f"""
+                [red]WARNING: The CSV file has {num_rows} rows. Loading this data and 
+                generating the graph database will take long time.
+                """
+            )
+
+            user_input_continue = Prompt.ask(
+                "Do you want to continue with the whole dataset? (y/n)",
+            )
+            if user_input_continue == "n":
+                sample_size = int(
+                    Prompt.ask(
+                        "Please enter the sample size",
+                    )
+                )
+                print(
+                    f"""
+                    [green]The graph database will be generated for {sample_size} 
+                    rows...
+                    """
+                )
+                csv_dataframe = csv_dataframe.sample(n=sample_size)
+            elif user_input_continue == "y":
+                print(
+                    """
+                    [green]The graph database will be generated for the whole dataset...
+                    """
+                )
+
+            csv_kg_chat_agent.csv_dataframe = csv_dataframe
+            csv_kg_chat_agent.csv_location = csv_location
+
+            build_kg_instructions = f"""
```

Comment:
the csv-kg-agent system prompts shouldn't be in this example. The system prompts should be internal to that agent, not exposed here.